cache/cache.cpp

```diff
@@ -260,6 +260,40 @@ static int dnet_cmd_cache_io_write(struct cache_manager *cache,
 	return err;
 }
 
+static ioremap::elliptics::dnet_lookup_response lookup_response_from_cache_item(const cache_item &it,
+                                                                                bool add_checksum,
+                                                                                struct dnet_node *n) {
+	using namespace ioremap::elliptics;
+
+	std::vector<unsigned char> json_checksum;
+	std::vector<unsigned char> data_checksum;
+
+	if (add_checksum) {
+		json_checksum.resize(DNET_CSUM_SIZE);
+		dnet_checksum_data(n, it.json->data(), it.json->size(), json_checksum.data(), json_checksum.size());
```

Comment:
`dnet_checksum_data` returns error code that should be checked and in case of error it should be returned to the client